dask/tests/test_highgraph.py

```diff
@@ -57,3 +59,51 @@ def test_cull():
 
     culled_by_y = hg.cull({"y"})
     assert dict(culled_by_y) == a
+
+
+@pytest.mark.parametrize("inject_dict", [True, False])
+def test_map_basic_layers(inject_dict):
+    """Check map_basic_layers() by injecting an inc() call"""
+
+    y = da.ones(3, chunks=(3,), dtype="int") + 40
+
+    def inject_inc(dsk):
+        assert isinstance(dsk, BasicLayer)
+        dsk = dict(dsk)
+        k = next(iter(dsk))
+        dsk[k] = (inc, dsk[k])
+        if inject_dict:
+            return dsk  # map_basic_layers() should automatically convert it to a `BasicLayer`
+        else:
+            return BasicLayer(dsk)
+
+    dsk = y.__dask_graph__()
+    y.dask = dsk.map_basic_layers(inject_inc)
+    layers = list(y.dask.layers.values())
+    assert type(layers[0]) == BasicLayer
+    assert type(layers[1]) == Blockwise
+    assert list(y.compute()) == [42] * 3
+
+
+@pytest.mark.parametrize("use_layer_map_task", [True, False])
+def test_map_tasks(use_layer_map_task):
+    """Check map_tasks() by injecting an +1 to the `40` literal"""
+    y = da.ones(3, chunks=(3,), dtype="int") + 40
+
+    def plus_one(tasks):
+        ret = []
+        for t in tasks:
+            if t == 40:
+                t += 1
+            ret.append(t)
+        return tuple(ret)
+
+    dsk = y.__dask_graph__()
+
+    if use_layer_map_task:
+        # Overwrite Blockwise.map_tasks with Layer.map_tasks
+        blockwise_layer = list(dsk.layers.values())[1]
+        blockwise_layer.map_tasks = partial(Layer.map_tasks, blockwise_layer)
```

Comment:
I'm a little confused as to what we're trying to test here

Reply:
Updated the description to more descriptive, hope it makes more sense?
```python 
    if use_layer_map_task:
        # In order to test the default map_tasks() implementation on a Blockwise Layer,
        # we overwrite Blockwise.map_tasks with Layer.map_tasks
        blockwise_layer = list(dsk.layers.values())[1]
        blockwise_layer.map_tasks = partial(Layer.map_tasks, blockwise_layer)
```